### DISCUSSION TITLE
Documentation on how to use new labels

### DISCUSSION THREAD

**Discussion by ericchansen** [tox-dev/tox#2406] [Q&A] [answered]
I want to create a new env that runs in almost the same way as the default env except it has more extras. I'll try to give a minimal example.

```
[tox]
envlist = py36, py37, py38, py39
minversion = 3.15

[testenv]
description = Invoke pytest to run automated tests
commands =
    pytest \
        --cov \
        --cov-append \
        --junitxml {toxworkdir}/junit.{envname}.xml \
        src/tests
        {posargs}
extras =
    testing
passenv =
    HOME
setenv =
    COVERAGE_FILE = {env:COVERAGE_FILE:{toxworkdir}/.coverage.{envname}}
    TOXINIDIR = {toxinidir}

[testenv:gitlab]
extras =
    gitlab
    testing
```

I want the `gitlab` env to do the same thing as the default env (run the tests on py36, py37, etc.), except also install a few more additional dependencies.

Based on the docs with the new version, I can't figure out how to make this happen.

**Comment by jugmac00** [comment, marked as answer]
This is untested, but should probably work:

```
[tox]
envlist = py37, py38
labels =
    gitlab = py37-gitlab, py38-gitlab

[testenv]
extras =
    testing
    gitlab: gitlab
commands = pytest
```

On gitlab you could then run `tox -m gitlab`.

Here, two features are combined.

The new labels feature, as outlined in https://tox.wiki/en/rewrite/config.html#labels and factors as described in https://tox.wiki/en/latest/config.html?highlight=factor#factors-and-factor-conditional-settings

Please note that `labels` are a feature of tox4 for which currently there is a beta available at https://pypi.org/project/tox/4.0.0b2/

**Reply by ericchansen** [reply]
This worked! I'm looking forward to the full release of version 4, along with plenty of docs so that I can figure all of this out myself.

**Reply by jugmac00** [reply]
https://twitter.com/gjbernat/status/1517806198770352133 :-)

**Reply by ssbarnea** [reply]
TBH, even now one needs to do extensive reasearch to figure out how to use the labels feature, even if the config item is documented... meaning, documented how to declare them, not how to use them. :p

**Reply by gaborbernat** [reply]
PR welcome 👍
